### ISSUE TITLE
Update from 0.3.0 to 0.4.0 fails to initialize, cannot configure

### ISSUE THREAD

**Issue by mickut** [JaccoR/hass-entso-e#169]
Updated to 0.4.0, entities do not update, cannot configure.

Trying to start the configuration flow results in "500 Server Error" dialog, system logs show error:

```
Error while setting up entsoe platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 364, in _async_setup_platform
    await asyncio.shield(awaitable)
  File "/config/custom_components/entsoe/sensor.py", line 105, in async_setup_entry
    for description in sensor_descriptions(currency = config_entry.options[CONF_CURRENCY]):
                                                      ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
KeyError: 'currency'
```

HA Core: 2024.8.3
Supervisor: 2024.08.0
HAOSS: 12.4

Raspberry Pi 4

**Comment by mickut** [comment]
Creating a new instance worked, old instance is broken. 

Any upgrader should keep a copy of both API key and possible value template safe to re-create the integration instance after update.

**Comment by Roeland54** [comment]
Thank you for reporting. I have created a fix for this and released v0.4.1 with this. 
Upgrading from v0.3.0 to v0.4.1 should just work without needing to recreate anything.
